### ISSUE TITLE
Don't warn about 'Redundant compare' in `(==)` and `(/=)` definitions

### ISSUE THREAD

**Issue by uzytkownik** [ndmitchell/hlint#774]
This is a minor improvement. It's often useful to have one implementation of compare and implement `(==)` in terms of it:

```
instance Eq MyType where
    l == r = compare l r == EQ

instance Ord MyType where
    compare l r = ...
```

However currently the hLint warns on the implementation of `(==)` and it seems that you cannot annotate instance method so you need to waive the hint for whole module.

**Comment by ndmitchell** [comment]
Agreed - would be a great thing to have. There's some kind of hack in there at the moment which tries not to suggest `==` if you are in the process of defining it (as you are there), but it's a fragile thing that doesn't seem to work great.
